File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int n = values.length;
        int[] maxLeftScore = new int[n];
        maxLeftScore[0] = values[0];  // Initialize the first element
        int maxScore = 0;

        for (int i = 1; i < n; i++) {
            int maxRScore = values[i] - i;  // Right score (values[j] - j)
            maxScore = Math.max(maxRScore + maxLeftScore[i - 1], maxScore);

            int maxLScr = values[i] + i;  // Left score (values[i] + i)
            maxLeftScore[i] = Math.max(maxLeftScore[i - 1], maxLScr);
        }
        return maxScore;
    }
}

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        max_leftscore = [0] * len(values)
        max_leftscore[0] = values[0]
        maxscore = 0

        for i in range(1, len(values)):
            max_rscore = values[i] - i
            maxscore = max(max_rscore + max_leftscore[i - 1], maxscore)

            maxlscr = values[i] + i 
            max_leftscore[i] = max(max_leftscore[i - 1], maxlscr)
        return maxscore
